centreon/www/include/Administration/myAccount/formMyAccount.php

```diff
@@ -418,7 +419,11 @@ function myReplace()
 $sessionKeyFreeze = 'administration-form-my-account-freeze';
 
 if ($form->validate()) {
-    updateContactInDB($centreon->user->get_id());
+    if ($cct['contact_auth_type'] === 'local') {
+        updateContactInDB($centreon->user->get_id());
+    } else {
+        updateLocalContactInDB($centreon->user->get_id());
```

Comment:
There's a small error in my last suggestion: it's update**Non**LocalContactInDB and not updateLocalContactInDB.
```suggestion
        updateNonLocalContactInDB($centreon->user->get_id());
```